Upgrade RAVEN to version 3.0

## Description

New version of Raven has been released. 

- Remove scale_factor and offset handling from python wrapper since it's handled in the model itself now.

Hum, getting a build error
```
g++ -Wno-deprecated -Dnetcdf                 -o Raven.exe  ./GlobalParams.o  ./TimeSeriesABC.o  ./Evaporation.o  ./HeatConduction.o  ./CustomOutput.o  ./Transport.o  ./VegetationParams.o  ./LandUseClass.o  ./SnowMeltRefreeze.o  ./Radiation.o  ./Decay.o  ./CapillaryRise.o  ./PartitionPrecip.o  ./ParseInput.o  ./SoilEvaporation.o  ./GroundwaterModel.o  ./ParseHRUFile.o  ./LatFlush.o  ./DepressionProcesses.o  ./ParseGWSWOverlapFile.o  ./ParsePropertyFile.o  ./HydroUnits.o  ./OrographicCorrections.o  ./SnowTempEvolve.o  ./StandardOutput.o  ./Convolution.o  ./ParseInitialConditionFile.o  ./SnowBalance.o  ./GreenAmpt.o  ./SparseMatrix.o  ./ParseEnsembleFile.o  ./Percolation.o  ./Baseflow.o  ./Solvers.o  ./GlacierProcesses.o  ./ModelInitialize.o  ./Advection.o  ./VegetationClass.o  ./ParseGWSWExchangeFile.o  ./ParseTimeSeriesFile.o  ./AquiferStack.o  ./TimeSeries.o  ./MassRouting.o  ./SoilClass.o  ./SubBasin.o  ./ParseGWPropsFile.o  ./Interflow.o  ./OpenWaterEvap.o  ./UpdateForcings.o  ./Transformation.o  ./Assimilate.o  ./SnowParams.o  ./Reservoir.o  ./Recharge.o  ./LateralExchangeABC.o  ./VegetationMovers.o  ./Drain.o  ./Flush.o  ./Sublimation.o  ./PotentialMelt.o  ./ForcingGrid.o  ./OverlapExchangeClass.o  ./GWGeometry.o  ./GWStressPeriodClass.o  ./TerrainClass.o  ./SoilProfile.o  ./HydroProcessABC.o  ./ParseGWGeometryFile.o  ./ParseLiveFile.o  ./RavenMain.o  ./StateVariables.o  ./LatAdvection.o  ./SoilParams.o  ./Infiltration.o  ./TransientParam.o  ./SnowAlbedo.o  ./ChannelXSect.o  ./ParseLib.o  ./UnitTesting.o  ./PrairieSnow.o  ./ProcessGroup.o  ./GWSolvers.o  ./CommonFunctions.o  ./Gauge.o  ./UTM_to_LatLong.o  ./HRUGroups.o  ./Diagnostics.o  ./ModelForcingGrids.o  ./Model.o  ./ModelParamCheck.o  ./CropGrowth.o  ./Abstraction.o  ./ModelEnsemble.o  ./IrregularTimeSeries.o  ./Forcings.o -L/usr/local -lnetcdf    -Wl,-stack_size,0x80000000,-stack_addr,0xf0000000 
/usr/bin/ld: unrecognized -a option `ck_size'
collect2: error: ld returned 1 exit status
```

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -1,6 +1,12 @@
 Changes
 =======
 
+0.11.x
+------
+* Update to Raven 3.0
+* Update to xclim 0.18
+* Update to xarray 0.16
+
 
 0.10.x (2020-03-09) Oxford
 --------------------------

--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ PYTHON_VERSION = 3.6
 
 # Choose Anaconda installer depending on your OS
 ANACONDA_URL = https://repo.anaconda.com/miniconda
-RAVEN_URL    = http://www.civil.uwaterloo.ca/jmai/raven/raven-rev254.zip
+RAVEN_URL    = http://www.civil.uwaterloo.ca/jmai/raven/Raven-rev263.zip
 RAVEN_SRC    = $(CURDIR)/src/RAVEN
 OSTRICH_URL  = http://www.civil.uwaterloo.ca/jmai/raven/Ostrich_2017-12-19_plus_progressJSON.zip
 OSTRICH_SRC  = $(CURDIR)/src/OSTRICH

--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ PYTHON_VERSION = 3.6
 
 # Choose Anaconda installer depending on your OS
 ANACONDA_URL = https://repo.anaconda.com/miniconda
-RAVEN_URL    = http://www.civil.uwaterloo.ca/jmai/raven/Raven-rev263.zip
+RAVEN_URL    = http://www.civil.uwaterloo.ca/jmai/raven/Raven-rev264.zip
 RAVEN_SRC    = $(CURDIR)/src/RAVEN
 OSTRICH_URL  = http://www.civil.uwaterloo.ca/jmai/raven/Ostrich_2017-12-19_plus_progressJSON.zip
 OSTRICH_SRC  = $(CURDIR)/src/OSTRICH

--- a/raven/models/base.py
+++ b/raven/models/base.py
@@ -535,7 +535,6 @@ class Raven:
                         # Check if any alternate variable name is in the file.
                         for alt_name in alt_names:
                             if alt_name in ds.data_vars:
-                                enc = ds[alt_name].encoding
                                 ncvars[var] = dict(var=var,
                                                    path=fn,
                                                    var_name=alt_name,

--- a/raven/models/base.py
+++ b/raven/models/base.py
@@ -541,8 +541,6 @@ class Raven:
                                                    var_name=alt_name,
                                                    dimensions=ds[alt_name].dims,
                                                    units=ds[alt_name].attrs.get("units"),
-                                                   scale_factor=enc.get("scale_factor"),
-                                                   add_offset=enc.get("add_offset")
                                                    )
 
                                 break

--- a/raven/models/global/global.rvc
+++ b/raven/models/global/global.rvc
@@ -1,5 +1,5 @@
 #########################################################################
-:FileType          rvc ASCII Raven 2.9
+:FileType          rvc ASCII Raven 3.0
 #
 # Initial conditions
 #-------------------

--- a/raven/models/rv.py
+++ b/raven/models/rv.py
@@ -378,7 +378,7 @@ class RVI(RV):
         self.latitude = None
         self.longitude = None
         self.run_index = 0
-        self.raven_version = '3.0 rev#263'
+        self.raven_version = '3.0 rev#264'
 
         self._run_name = 'run'
         self._start_date = None

--- a/raven/models/rv.py
+++ b/raven/models/rv.py
@@ -288,11 +288,9 @@ class RavenNcData(RV):
     def linear_transform(self):
         """A sequence of two values: multiplicative factor and additive offset."""
         lt = self._linear_transform
-        if lt is not None or self.scale_factor is not None or self.add_offset is not None:
+        if lt is not None:
             slope, intercept = lt or (1, 0)
-            sf = 1 if self.scale_factor is None else self.scale_factor
-            offset = 0 if self.add_offset is None else self.add_offset
-            return ":LinearTransform {:.15f} {:.15f}".format(slope * sf, offset * slope + intercept)
+            return ":LinearTransform {:.15f} {:.15f}".format(slope, intercept)
 
     @linear_transform.setter
     def linear_transform(self, value):
@@ -380,7 +378,7 @@ class RVI(RV):
         self.latitude = None
         self.longitude = None
         self.run_index = 0
-        self.raven_version = '2.9 rev#254'
+        self.raven_version = '3.0 rev#263'
 
         self._run_name = 'run'
         self._start_date = None

--- a/tests/test_emulators.py
+++ b/tests/test_emulators.py
@@ -233,10 +233,10 @@ class TestGR4JCN:
 
     def test_version(self):
         model = Raven()
-        assert model.version == "2.9.2"
+        assert model.version == "2.9.9"
 
         model = GR4JCN()
-        assert model.version == "2.9.2"
+        assert model.version == "2.9.9"
 
     def test_parallel_params(self):
         ts = TESTDATA["raven-gr4j-cemaneige-nc-ts"]

--- a/tests/test_emulators.py
+++ b/tests/test_emulators.py
@@ -233,10 +233,10 @@ class TestGR4JCN:
 
     def test_version(self):
         model = Raven()
-        assert model.version == "2.9.9"
+        assert model.version == "3.0"
 
         model = GR4JCN()
-        assert model.version == "2.9.9"
+        assert model.version == "3.0"
 
     def test_parallel_params(self):
         ts = TESTDATA["raven-gr4j-cemaneige-nc-ts"]

